service/history/workflow/update/registry.go

```diff
@@ -195,10 +196,18 @@ func (r *registry) Find(ctx context.Context, id string) (*Update, bool) {
 	return r.findLocked(ctx, id)
 }
 
-func (r *registry) TerminateUpdates(_ context.Context, _ EventStore) {
-	// TODO (alex-update): implement
-	// This method is not implemented and update API callers will just timeout.
-	// In future, it should remove all existing updates and notify callers with better error.
+// CancelIncomplete cancels all incomplete updates in the registry:
+//   - updates in stateAdmitted, stateRequested, or stateSent are rejected,
+//   - updates in stateAccepted are ignored (see CancelIncomplete() in update.go for details),
+//   - updates in stateCompleted are ignored.
+func (r *registry) CancelIncomplete(ctx context.Context, reason CancelReason, eventStore EventStore) error {
+	incompleteUpdates := r.enumerate(func(u *Update) bool { return u.isIncomplete() })
```

Comment:
It looks like you filter all the updates which are complete. What is the reason to call them `incompleUpdates`

Reply:
`isIncomplete` returns true if update is not complete.

Reply:
Ah, yes, thank you. Probably too tired to review code.